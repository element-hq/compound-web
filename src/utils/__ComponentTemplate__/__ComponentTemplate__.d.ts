import React, { PropsWithChildren } from "react";
type __ComponentTemplate__Props<C extends React.ElementType> = {
    as?: C;
    className?: string;
} & React.ComponentPropsWithoutRef<C>;
export declare const __ComponentTemplate__: <C extends React.ElementType<any> = "p">({ as, children, className, ...props }: React.PropsWithChildren<__ComponentTemplate__Props<C>>) => React.JSX.Element;
export {};
