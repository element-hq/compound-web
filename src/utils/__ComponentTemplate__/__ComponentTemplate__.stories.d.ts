import React from "react";
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, {
    as?: React.ElementType<any> | undefined;
    className?: string | undefined;
} & Omit<any, "ref"> & {
    children?: React.ReactNode;
}>;
export default _default;
export declare const Round: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, {
    as?: React.ElementType<any> | undefined;
    className?: string | undefined;
} & Omit<any, "ref"> & {
    children?: React.ReactNode;
}>;
