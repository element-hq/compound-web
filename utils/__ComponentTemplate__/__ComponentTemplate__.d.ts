import React from "react";
export declare const __ComponentTemplate__: React.ForwardRefExoticComponent<{
    className?: string | undefined;
} & React.HTMLAttributes<HTMLDivElement> & React.RefAttributes<HTMLDivElement>>;
//# sourceMappingURL=__ComponentTemplate__.d.ts.map