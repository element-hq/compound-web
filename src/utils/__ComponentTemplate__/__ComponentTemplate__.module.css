/*
Copyright %YEAR% %COPYRIGHT_HOLDER%

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

.root {
  /* TODO: Update the class name to something related to the component name */
  all: initial;
}
