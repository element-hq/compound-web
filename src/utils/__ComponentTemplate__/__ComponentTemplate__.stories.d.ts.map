{"version":3,"file":"__ComponentTemplate__.stories.d.ts","sourceRoot":"","sources":["../../../../src/utils/__ComponentTemplate__/__ComponentTemplate__.stories.tsx"],"names":[],"mappings":"AAgBA,OAAO,KAAK,MAAM,OAAO,CAAC;;;;;;;AAK1B,wBAKiD;AAMjD,eAAO,MAAM,KAAK;;;;;EAAoB,CAAC"}