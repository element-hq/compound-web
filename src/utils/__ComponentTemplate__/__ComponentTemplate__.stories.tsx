/*
Copyright %YEAR% %COPYRIGHT_HOLDER%

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { Meta, StoryFn } from "@storybook/react";

import { __ComponentTemplate__ as __ComponentTemplate__Component } from "./__ComponentTemplate__";

export default {
  title: "__ComponentTemplate__",
  component: __ComponentTemplate__Component,
  argTypes: {},
  args: {},
} as Meta<typeof __ComponentTemplate__Component>;

const Template: StoryFn<typeof __ComponentTemplate__Component> = (args) => (
  <__ComponentTemplate__Component {...args} />
);

export const Round = Template.bind({});
Round.args = {};
Round.parameters = {
  design: {
    type: "figma",
    url: "%FIGMA_URL%",
  },
};
