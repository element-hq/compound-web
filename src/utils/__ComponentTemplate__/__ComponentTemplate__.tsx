/*
Copyright %YEAR% %COPYRIGHT_HOLDER%

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classnames from "classnames";
import React, { forwardRef, PropsWithChildren } from "react";
import styles from "./__ComponentTemplate__.module.css";

type __ComponentTemplate__Props = {
  className?: string;
} & React.HTMLAttributes<HTMLDivElement>;

export const __ComponentTemplate__ = forwardRef<
  HTMLDivElement,
  __ComponentTemplate__Props
>(function __ComponentTemplate__(
  {
    children,
    className,
    ...props
  }: PropsWithChildren<__ComponentTemplate__Props>,
  ref,
) {
  // TODO: Update the class name to something related to the component name
  const classes = classnames(styles.root, className);
  return (
    <div {...props} className={classes} ref={ref}>
      {children}
    </div>
  );
});
