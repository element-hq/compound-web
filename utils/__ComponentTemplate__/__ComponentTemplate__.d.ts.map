{"version":3,"file":"__ComponentTemplate__.d.ts","sourceRoot":"","sources":["../../../src/utils/__ComponentTemplate__/__ComponentTemplate__.tsx"],"names":[],"mappings":"AAiBA,OAAO,KAAwC,MAAM,OAAO,CAAC;AAO7D,eAAO,MAAM,qBAAqB;;+EAkBhC,CAAC"}