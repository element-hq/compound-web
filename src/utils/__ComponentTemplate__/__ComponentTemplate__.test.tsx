/*
Copyright %YEAR% %COPYRIGHT_HOLDER%

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

import { __ComponentTemplate__ } from "./__ComponentTemplate__";

describe("__ComponentTemplate__", () => {
  it("renders", () => {
    const { asFragment } = render(<__ComponentTemplate__ />);
    expect(asFragment()).toMatchSnapshot();
  });
});
