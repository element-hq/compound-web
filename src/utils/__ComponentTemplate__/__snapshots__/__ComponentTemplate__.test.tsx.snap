// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__ComponentTemplate__ renders 1`] = `
<DocumentFragment>
  <div
    class="root"
  />
</DocumentFragment>
`;
